### PR TITLE
rename controller & service to be consistent with other net-* repos

### PR DESCRIPTION
**Pull request by dprotaso**
Fixes: https://github.com/knative/networking/issues/448

### CODE REVIEW

File: config/controller.yaml [hunk 1]
```diff
@@ -82,7 +82,7 @@ spec:
 apiVersion: v1
 kind: Service
 metadata:
-  name: challenger
+  name: net-http01-controller
```
Comment (dprotaso) [inline config/controller.yaml:85]: `challenger` I don't see this name being used anywhere in the code base

Comment (dprotaso) [reply config/controller.yaml:85]: Brought up whether this is actually used anywhere

**Review by ZhiminXiang — APPROVED** [review]
/lgtm

**Pull request merged**